File: Codechef codes/CSUB.py
T = int(raw_input())
while(T>0):
	N = int(raw_input())
	S = str(raw_input())
	count = 0

	for i in range(len(S)):
		if S[i] == '1':
			count = count+1

	print (count*(count+1))/2
	T = T-1

File: Codechef codes/CSUB.c
#include<stdio.h>

int main()
{
	long long int T;
	scanf("%lld",&T);
	
	while(T--)
	{
		long long int N,count=0,i;
		scanf("%lld",&N);
		
		char S[N];
		scanf("%s",S);
		for(i=0;i<N;i++)
			if(S[i] == '1')
				count++;

		printf("%lld\n",(count*(count+1)/2));
	}
	return 0;
}
